### ISSUE TITLE
how to use tf.contrib.framework.fuse_op to fuse graph

### ISSUE THREAD

**Issue by wangyunxiaa** [tensorflow/tensorflow#28038]
Hello,
I want to use tf.contrib.framework.fuse_op to fuse graph to fuse my graph for faster inference,but when i call the api 
`new_graphdef = tf.contrib.framework.fuse_op(
                                           graph_def,
                                           input_nodes=["ac_input"],
                                           output_nodes=["Inference/input_tansfer/mul"],
                                           output_dtypes=[tf.float32],
                                           output_quantized=False,
                                           op_name="trans_fused",
                                           op_type="add_mul"
                                       )`
I got an error
`File "build_fusedop.py", line 1006, in main
    op_type="add_mul" 
  File "/home/yx.wang/anaconda3/envs/py27tfgpu/lib/python2.7/site-packages/tensorflow/contrib/framework/python/framework/graph_util.py", line 115, in fuse_op
    new_node.attr["_output_types"].list.type[:] = output_dtypes
TypeError: tf.float32 has type DType, but expected one of: int, long`

I cannot find the detail deacription of the parameters, the source code as followings:
`graph_def: A graph_pb2.GraphDef proto.
input_nodes: input nodes to the subgraph to be fused.
output_nodes: output nodes to the subgraph to be fused.
output_dtypes: A list of output datatypes for the custom op
output_quantized: A boolean flag that indicates if output is quantized
op_name: fused op name.
op_type: fused op type.` 

what is the correct input of the api?

**Comment by wangyunxiaa** [comment]
Sorry, I found  the test example and changed my code to 
`input_list = ["ac_input"]
                output_list = ["Inference/Reshape"]
                default_type = types_pb2.DT_FLOAT
                new_graph = fuse_op(
                                    graph_def=graph_def,
                                    input_nodes=input_list,
                                    output_nodes=output_list,
                                    output_dtypes=[default_type for _ in output_list],
                                    output_quantized=False,
                                    op_name="trans_fused",
                                    op_type="CustomTfTransOp")`
and del the 123 line of https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/contrib/framework/python/framework/graph_util.py `assert len(name_to_node[n].input) == 1` , but the way, why do the assert ? 

with above changes, I can return and save the new_graph, But when load the saved graph I got an error 
`tensorflow.python.framework.errors_impl.NotFoundError: Op type not registered 'CustomTfTransOp' in binary running on ac2690. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
`
what does the fused operator do?  If I want to fuse the add and mul operation together, Should I add  an op do this? what is the mean of "fuse_op"?

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by wangyunxiaa** [comment]
@jvishnuvardhan ok, But I think the function of this api is to merge existing calculations to a new op, why the new op doesnot exit? if I should add a new op with the new operation I don't need to use this  api

**Comment by jvishnuvardhan** [comment]
@wangyunxiaa If you have a solution, you could raise a PR and merge your contributions to TF repository. Please let me know if you want to contribute. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will open a new issue. Thanks!
